src/applications/disability-benefits/526EZ/components/ConfirmationPoll.jsx

```diff
@@ -0,0 +1,192 @@
+import React from 'react';
+import Raven from 'raven-js';
+import { connect } from 'react-redux';
+
+import LoadingIndicator from '@department-of-veterans-affairs/formation/LoadingIndicator';
+
+import get from '../../../../platform/utilities/data/get';
+import { apiRequest } from '../../../../platform/utilities/api';
+
+import ConfirmationPage from '../containers/ConfirmationPage';
+
+export const submissionStatuses = {
+  // Statuses returned by the API
+  pending: 'submitted', // Submitted to EVSS, waiting response
+  retry: 'retrying',
+  succeeded: 'received', // Submitted to EVSS, received response
+  exhausted: 'exhausted', // EVSS is down or something; ran out of retries
+  failed: 'non_retryable_error', // EVSS responded with some error
+  // When the api serves a failure
+  apiFailure: 'apiFailure'
+};
+
+const terminalStatuses = [
+  submissionStatuses.succeeded,
+  submissionStatuses.exhausted,
+  submissionStatuses.retry,
+  submissionStatuses.failed
+];
+
+const successMessage = (claimId) => (
```

Comment:
It feels a little weird to me to have the messages in here. This component is mostly focused on polling, but with these UI snippets thrown in. Might be cleaner to have all the UI code in ConfirmationPage and leave this component to do polling and pass whatever status you need to ConfirmationPage.

Reply:
Good point again. This is left over from when I had the component hierarchy swapped around. Definitely makes more sense to have all the UI in the one component now. :+1: